libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -390,70 +390,105 @@ class WingRestApi extends Construct {
+            exposeHeaders: props.cors.exposeHeaders
+              ? props.cors.exposeHeaders
+              : [],
+            allowCredentials: props.cors.allowCredentials
+              ? props.cors.allowCredentials
+              : false,
+            // added maxAge
+            maxAge: props.cors.maxAge ? props.cors.maxAge : 0,
+          },
+        }
+      : {};
 
-    this.api = new ApiGatewayRestApi(this, "api", {
+    this.api = new Apigatewayv2Api(this, "api", {
       name: ResourceNames.generateName(this, NAME_OPTS),
-      // Lazy generation of the api spec because routes can be added after the API is created
+      protocolType: "HTTP",
       body: Lazy.stringValue({
         produce: () => {
-          const injectGreedy404Handler = (openApiSpec: OpenApiSpec) => {
-            openApiSpec.paths = {
-              ...openApiSpec.paths,
-              ...defaultResponse,
-            };
-            return openApiSpec;
-          };
-          return JSON.stringify(injectGreedy404Handler(props.apiSpec));
+          props.apiSpec = "random assignment so compiler doesn't complain";
+          return JSON.stringify(OAS30);
         },
       }),
+      ...apigwCorsConfig,
     });
 
-    this.deployment = new ApiGatewayDeployment(this, "deployment", {
-      restApiId: this.api.id,
-      lifecycle: {
-        createBeforeDestroy: true,
-      },
-      triggers: {
-        // Trigger redeployment when the api spec changes
-        redeployment: Fn.sha256(this.api.body),
-      },
-    });
-
-    this.stage = new ApiGatewayStage(this, "stage", {
-      restApiId: this.api.id,
-      stageName: STAGE_NAME,
-      deploymentId: this.deployment.id,
+    this.stage = new Apigatewayv2Stage(this, "stage", {
+      apiId: this.api.id,
+      name: "$default",
+      autoDeploy: true,
     });
 
-    //should be exported from here, otherwise won't be mapped to the right token
-    this.url = this.stage.invokeUrl;
+    this.url = `https://${this.api.id}.execute-api.${this.region}.amazonaws.com`;
   }
 
-  /**
-   * Add an endpoint to the API
-   * @param path Path of the endpoint
-   * @param method Method of the endpoint
-   * @param handler Lambda function to handle the endpoint
-   * @returns OpenApi spec extension for the endpoint
-   */
   public addEndpoint(path: string, method: string, handler: Function) {
     const endpointExtension = this.createApiSpecExtension(handler);
     this.addHandlerPermissions(path, method, handler);
     return endpointExtension;
   }
 
-  /**
-   * Creates a OpenApi extension object for the endpoint and handler
-   * @param handler Lambda function to handle the endpoint
-   * @returns OpenApi extension object for the endpoint and handler
-   */
   private createApiSpecExtension(handler: Function) {
     const extension = {
       "x-amazon-apigateway-integration": {
         uri: `arn:aws:apigateway:${this.region}:lambda:path/2015-03-31/functions/${handler.arn}/invocations`,
```

Comment:
I believe this might need to change to `arn:aws:lambda:us-east-2:123456789012:function:HelloWorld` according to https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-open-api.html

Reply:
I updated the code. The `_addToSpec` method fails to update the spec of the ApiGw for some reasons.

Reply:
Btw, I've just tried to export an OAS30 and the format was indeed correct.

Reply:
Ok, I did some tests and both formats seem to work, so we can just use the lambda function ARN since that's shorter.